packages/react/src/use-table/helpers.tsx

```diff
@@ -336,27 +336,32 @@ const maybeGetDefaultDisplayFieldMetadata = (fieldMetadataTree: TableSpec["field
   }
 };
 
-export const fieldMetadataArrayToFieldMetadataTree = (fieldMetadataArray: FieldMetadataFragment[]) => {
+/**
+ * This turns the flat list of field metadata returned by the backend into a nested tree of field metadata.
+ * The paths in the object are precomputed to a fixed max depth, for easy lookups later in the table code
+ */
+export const fieldMetadataArrayToFieldMetadataTree = (fieldMetadataArray: FieldMetadata[], depth = 0) => {
```

Comment:
I didn't want to mess with the useTable stuff too much, this method builds out a tree that has the possible select paths that a table can take, we have to stop pre-expanding these paths out at a certain depth, I think 3 is sufficient for how AutoTable works